superset/db_engine_specs/bigquery.py

```diff
@@ -380,18 +381,47 @@ def build_sqlalchemy_uri(
 
     @classmethod
     def get_parameters_from_uri(
-        cls, uri: str, encrypted_extra: Optional[Dict[str, str]] = None
+        cls,
+        uri: str,
+        encrypted_extra: Optional[Dict[str, Any]] = None,
     ) -> Any:
         value = make_url_safe(uri)
 
         # Building parameters from encrypted_extra and uri
         if encrypted_extra:
+            try:
+                encrypted_extra["credentials_info"]["private_key"] = PASSWORD_MASK
+            except KeyError:
+                pass
+
             # ``value.query`` needs to be explicitly converted into a dict (from an
             # ``immutabledict``) so that it can be JSON serialized
             return {**encrypted_extra, "query": dict(value.query)}
 
         raise ValidationError("Invalid service credentials")
 
+    @classmethod
+    def update_encrypted_extra(
```

Comment:
What do you think about adding a unit test to bigquery and gsheets. An actual test for this has an integration test for the API would be nice also

Reply:
Working on it!